Tests/CacheAdvanceTests/CacheAdvanceTests.swift

```diff
@@ -24,6 +24,47 @@ final class CacheAdvanceTests: XCTestCase {
 
     // MARK: Behavior Tests
 
+    func test_isEmpty_returnsTrueWhenCacheIsEmpty() throws {
+        let cache = try createCache()
+
+        XCTAssertTrue(try cache.isEmpty())
+    }
+
+    func test_isEmpty_returnsFalseWhenCacheHasASingleMessage() throws {
+        let message: TestableMessage = "This is a test"
+        let cache = try createCache(messages: [message])
+        try cache.append(message: message)
+
+        XCTAssertFalse(try cache.isEmpty())
+    }
+
+    func test_isEmpty_returnsFalseWhenOpenedOnCacheThatHasASingleMessage() throws {
+        let message: TestableMessage = "This is a test"
+        let cache = try createCache()
+        try cache.append(message: message)
+
+        let sameCache = try createCache(zeroOutExistingFile: false)
+
+        XCTAssertFalse(try sameCache.isEmpty())
+    }
+
+    func test_isEmpty_returnsFalseWhenCacheThatDoesNotOverwriteIsFull() throws {
+        let message: TestableMessage = "This is a test"
+        let cache = try createCache(messages: [message])
```

Comment:
It could be nice to pass `shouldOverwriteOldMessages: false` (which I assume is the default) to make the intent more clear.

Reply:
it is indeed the default. I'll think I'll remove the default parameter in a separate PR. The defaults and parameter names as-is are... a bit less intuitive than I originally imagined.